docs/root/intro/arch_overview/load_balancing.rst

```diff
@@ -108,7 +108,8 @@ destination address of the incoming connection was before the connection was red
 Envoy. New destinations are added to the cluster by the load balancer on-demand, and the cluster
 :ref:`periodically <config_cluster_manager_cluster_cleanup_interval_ms>` cleans out unused hosts
 from the cluster. No other :ref:`load balancing type <config_cluster_manager_cluster_lb_type>` can
-be used with original destination clusters.
+be used with original destination clusters. Envoy can pick up the original destination either from 
+the kernel or from a HTTP header called ``x-envoy-original-destination-host``.
```

Comment:
Can we potentially have a short section on this header somewhere, with an example of what the header looks like. We should clarify it must be a fully resolved IP address, and not a DNS address. We can link to this section from the service discovery docs you also updated.

Reply:
I could not find any better place to add this header. Possibly in future we could add a section listing all http headers and what they mean so that it is just easier to read all of them at one place. I clarified about fully resolved IP address. The link to service discovery section already exists in the top the section.